includes/wc-formatting-functions.php

```diff
@@ -260,18 +260,26 @@ function wc_format_decimal( $number, $dp = false, $trim_zeros = false ) {
 	$decimals = array( wc_get_price_decimal_separator(), $locale['decimal_point'], $locale['mon_decimal_point'] );
 
 	// Remove locale from string.
-	if ( ! is_float( $number ) ) {
+	if (  gettype( $number ) === 'string' && ! is_numeric( $number ) ) {
 		$number = str_replace( $decimals, '.', $number );
 		$number = preg_replace( '/[^0-9\.,-]/', '', wc_clean( $number ) );
 	}
 
+	// Trim left zeros.
+	if (  gettype( $number ) === 'string' && is_numeric( $number ) ) {
```

Comment:
So many conditionals, just to remove zeros on left... Kind of thing that makes think if it's really necessary.